app/models/jobs_json_request_handler.rb

```diff
@@ -26,42 +26,44 @@ def clusters
     end
   end
 
-  def job_info_enumerator(cluster)
+  def job_info_enumerator(cluster) 
     if filter.user?
       cluster.job_adapter.info_where_owner_each(OodSupport::User.new.name, attrs: JOB_ATTRS)
     else
       cluster.job_adapter.info_all_each(attrs: JOB_ATTRS)
     end
-  end
+  end 
 
   def render
     response.content_type = Mime[:json]
 
     errors = []
-    count = 0
-    response.stream.write '{"data":[' # data is now an array of arrays']}'
+    count = 0 
 
     clusters.each_with_index do |cluster|
       begin
         job_info_enumerator(cluster).each_slice(3000) do |jobs|
           jobs = convert_info_filtered(filter.apply(jobs), cluster)
+          
+          jobs.each do |job|
 
-          response.stream.write "," if count > 0
-          response.stream.write jobs.to_json
+           response.stream.write job.to_json + "\n"
+          end
```

Comment:
It would change the client side logic, but I think this might be preferable:

    response.stream.write jobs.to_json + "\n"

Because what you are doing right now is splitting up this array into separate rows, one per job, and then on the client side you have the problem now of throttling the table re-drawing, because you don't want to redraw on every job add, which is what is happening right now.

If one row contains all the jobs from a cluster, or the slice of 3000 jobs, then your client side code can process this as a single ndjson object (one array of these jobs) which is faster since you do one draw for this chunk.

i.e. client side

```
table.rows.add(jobs).draw();
```

Client side checking whether this is an array or not is faster than whether the pbsid key exists.